src/kafka-file/languageservice/inlineCommandActivation.ts

```diff
@@ -0,0 +1,46 @@
+import * as vscode from "vscode";
+import { ConsumerLaunchState } from "../../client";
+import { StartConsumerCommandHandler, StopConsumerCommandHandler } from "../../commands/consumers";
+import { ProduceRecordCommandHandler } from "../../commands/producers";
+import { ConsumerLaunchStateProvider } from "./kafkaFileLanguageService";
+import { ConsumerBlock, KafkaFileDocument, Node, NodeKind, ProducerBlock } from "./parser/kafkaFileParser";
+
+function isProducerBlock(node: Node): node is ProducerBlock {
+    return node.kind === NodeKind.producerBlock;
+}
+  
+function isConsumerBlock(node: Node): node is ConsumerBlock {
+    return node.kind === NodeKind.consumerBlock;
+}
+
+export async function executeInlineCommand(kafkaFileDocument: KafkaFileDocument, clusterId: string, consumerLaunchStateProvider: ConsumerLaunchStateProvider) {
+    const editor = vscode.window.activeTextEditor;
+    const curPos = editor!.selection.active;
+    let node = kafkaFileDocument.findNodeBefore(curPos);
+    if (node === undefined) {
+        return;
+    }
+    if (![NodeKind.producerBlock, NodeKind.consumerBlock].includes(node.kind)) {
+        if (node.parent === undefined) {
+            return;
+        }
+        node = node.parent;
+    }
+    
+    let command: string | undefined;
+    let commandArguments: any[] | undefined;
+    
+    if (isProducerBlock(node)) {
+        command = ProduceRecordCommandHandler.commandId;
+        commandArguments = [node.createCommand(clusterId), 1];
+    }else if (isConsumerBlock(node)) {

```

Comment:
`} else if`

Reply:
Fixed, it might be nice to add prettier, and lint rules to avoid this.
Anyway, nice catch :)